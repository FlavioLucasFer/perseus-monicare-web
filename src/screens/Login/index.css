@import '~antd/dist/antd.css';

.container-login {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    align-content: space-between;
    width: 100vw;
    height: 100vh;
    background-color: #eee;
  }

  .user-login {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    padding: 40px 11px;
    /* height: 240px; */
    width: 400px;
    border-radius: 15px;
    text-align: left;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
    overflow: hidden;
    background-color: #fff;
    margin-top: 20px;
  }

  Image {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 50%;
    height: 50%;
  }

  a {
    align-self: center;
    color: #070653;
  }

  