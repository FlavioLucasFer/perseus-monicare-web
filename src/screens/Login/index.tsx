import 'screens/Login/index.css';
import React from 'react';
import { Form, Input, Button } from 'antd';

import Logo from 'components/Logo';
import { useNavigate } from 'react-router-dom';

const Login = () => {
	const navigate = useNavigate();

	const onFinish = (values: any) => {
		console.log('Success:', values);
		navigate('/patients/measurements');
	};

	const onFinishFailed = (errorInfo: any) => {
		console.log('Failed:', errorInfo);
	};

	const { Item: FormItem } = Form;
	const { Password: InputPassword } = Input;

	return (
		<div className='container-login'>
			<Logo className='logo' />

			<div className='user-login'>
				<Form
					name='userLogin'
					onFinish={onFinish}
					onFinishFailed={onFinishFailed}
					layout='vertical'
				>
					<FormItem
						label='Nome de Usuário:'
						name='username'
						rules={[
							{
								required: true,
								message: 'Digite seu nome de usuário'
							}
						]}
					>
						<Input />
					</FormItem>

					<FormItem
						label='Senha:'
						name='password'
						rules={[
							{
								required: true,
								message: 'Digite sua senha'
							}
						]}
					>
						<InputPassword />
					</FormItem>

					<FormItem>
						<Button
							type='primary'
							htmlType='submit'
							block
							style={{
								background: '#FC9A9A',
								borderColor: '#FF1616',
								fontSize: 18,
								color: 'black',
								height: 'auto',
							}}
						>
							Entrar
						</Button>
					</FormItem>

				</Form>

				<a className='password-recovery-link' href=''>Esqueci minha senha</a>
			</div>


		</div>
	);
}

export default Login;