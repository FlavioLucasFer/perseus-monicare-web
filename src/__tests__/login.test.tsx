import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Routes } from 'react-router-dom';
import Login from 'screens/Login';

afterEach(cleanup);
const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'), 
	useNavigate: () => mockedUseNavigate,
}));

describe('Login Screen', () => {
	it('Should render properly', () => {
		const wrapper = shallow(<Login/>);
		expect(wrapper.debug()).toMatchSnapshot();

	}); 

	// const linkElement = screen.getByText('Esqueci minha senha');
	// expect(linkElement).toBeInTheDocument();

    
	// const {debug, getByText} = render(<Login/>)
	// expect(getByText('Entrar').tagName).toBe('BUTTON')
	// debug()
});