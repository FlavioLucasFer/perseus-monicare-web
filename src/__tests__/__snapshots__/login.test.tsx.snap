// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Should render properly 1`] = `
"<div className=\\"container-login\\">
  <Logo className=\\"logo\\" />
  <div className=\\"user-login\\">
    <ForwardRef(InternalForm) name=\\"userLogin\\" onFinish={[Function: onFinish]} onFinishFailed={[Function: onFinishFailed]} layout=\\"vertical\\">
      <FormItem label=\\"Nome de UsuÃ¡rio:\\" name=\\"username\\" rules={{...}}>
        <Input type=\\"text\\" />
      </FormItem>
      <FormItem label=\\"Senha:\\" name=\\"password\\" rules={{...}}>
        <Password action=\\"click\\" visibilityToggle={true} iconRender={[Function: iconRender]} />
      </FormItem>
      <FormItem>
        <Button type=\\"primary\\" htmlType=\\"submit\\" block={true} style={{...}}>
          Entrar
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
    <a className=\\"password-recovery-link\\" href=\\"\\">
      Esqueci minha senha
    </a>
  </div>
</div>"
`;
